ipsets/ipsets.go

```diff
@@ -792,6 +820,27 @@ func (s *IPSets) ApplyDeletions() {
 	s.gaugeNumIpsets.Set(float64(len(s.ipSetIDToIPSet)))
 }
 
+// tryTempIPSetDeletions tries to delete any temporary IP sets found by the last resync.
+func (s *IPSets) tryTempIPSetDeletions() {
+	s.pendingTempIPSetDeletions.Iter(func(item interface{}) error {
+		setName := item.(string)
+		logCxt := s.logCxt.WithField("setName", setName)
+		if s.existingIPSetNames.Contains(setName) {
+			logCxt.Info("Deleting IP set.")
+			if err := s.deleteIPSet(setName); err != nil {
+				// Log and carry on; we'll try again in ApplyDeletions().
+				logCxt.WithError(err).Warning("Failed to delete temporary IP set. Will retry...")
+			} else {
+				// Success! Remove from the main pending deletions set too.
+				logCxt.WithField("setName", setName).Info("Successfully removed left-over temporary IP set.")
+				s.pendingIPSetDeletions.Discard(item)
+			}
+		}
```

Comment:
Should there be an else branch here too, that also removes from `s.pendingTempIPSetDeletions` (however that is done)?

Reply:
No, that's handled by returning `set.RemoveItem`

Reply:
Ah OK, I see now - sorry for the noise!